### PR TITLE
Remove 32-bit Mac and non-ARC support

### LINKED ISSUE #755: Remove 32-bit Mac Support
I don't think anyone is using it anymore now that Office is 64-bit only on Mac.

### PR DESCRIPTION
**Pull request by RPangrle**
There is one customer we added 32-bit support for has since gone 64-bit only. macOS has run only on 64-bit machines since 10.7 came out 5 years ago, so it's time to drop 32-bit Mac support.

Fixes #755

### CODE REVIEW

**Comment by unpluggedk** [comment]
The following files are removed in #860 
ADAL/src/ADInstanceDiscovery.h, .m, 
ADAL/tests/ADInstanceDiscoveryTests.m 

ADAL/tests/ADAuthorityValidationTests.m replaces ADInstanceDiscoveryTests.m and still contains the changes applied in this PR. 

Please remove the first 2 mentioned and apply the removal of the non-ARC support on the third. Otherwise, please proceed and open an issue for me to follow up on,

Thanks!

**Comment by RPangrle** [comment]
I removed the SAFE_ARC macros themselves, so I'll have to fix up any extra files before I can check in.

File: ADAL/src/cache/ADTokenCacheItem+Internal.h [hunk 1]
```diff
@@ -27,12 +27,14 @@
 @class ADAuthenticationError;
 @class ADAuthenticationResult;
 
-@interface ADTokenCacheItem (Internal)
+@interface ADTokenCacheItem ()
+
+@property (readwrite) NSMutableDictionary * additionalClient;
+@property (readonly) NSDictionary * additionalServer;
```
Comment (jasoncoolmax) [inline ADAL/src/cache/ADTokenCacheItem+Internal.h]: why do we put these two in class extension but the rest in class category? :)

File: ADAL/src/request/ADAcquireTokenSilentHandler.m [hunk 2]
```diff
@@ -80,19 +79,10 @@ - (void)getToken:(ADAuthenticationCallback)completionBlock
 
 - (void)dealloc
 {
```
Comment (jasoncoolmax) [inline ADAL/src/request/ADAcquireTokenSilentHandler.m]: shall we remove the dealloc fun as you did in other places? :)

File: ADAL/src/ui/mac/ADAuthenticationViewController.m [hunk 3]
```diff
@@ -45,6 +45,13 @@ static NSRect _CenterRect(NSRect rect1, NSRect rect2)
 
 
 @interface ADAuthenticationViewController ( ) <WebResourceLoadDelegate, WebPolicyDelegate, WebFrameLoadDelegate, NSWindowDelegate>
+{
+    // In the legacy ObjC runtime (which is what we're stuck at for 32-bit Mac builds)
+    // you can't define the ivars of a class in the implementation file.
```
Comment (jasoncoolmax) [inline ADAL/src/ui/mac/ADAuthenticationViewController.m]: do we still need these comments? :)

File: ADAL/src/ADInstanceDiscovery.m [hunk 4]
```diff
@@ -0,0 +1,369 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (jasoncoolmax) [inline ADAL/src/ADInstanceDiscovery.m]: why do we still need this file? :)

File: ADAL/src/ui/mac/ADAuthenticationViewController.m (hunk 3 above)
Comment (RPangrle) [inline ADAL/src/ui/mac/ADAuthenticationViewController.m]: Possibly not.

File: ADAL/src/cache/ADTokenCacheItem+Internal.h (hunk 1 above)
Comment (RPangrle) [inline ADAL/src/cache/ADTokenCacheItem+Internal.h]: You can only add properties in an extension, unless you provide some other mechanism for their storage. Adding them like this means they'll get auto-synthesized and I don't have to bother providing implementation for them. :)

File: ADAL/src/ADInstanceDiscovery.m (hunk 4 above)
Comment (RPangrle) [inline ADAL/src/ADInstanceDiscovery.m]: No, it was supposed to be deleted in the merge :-/

File: ADAL/src/cache/ADTokenCacheItem+Internal.h (hunk 1 above)
Comment (jasoncoolmax) [inline ADAL/src/cache/ADTokenCacheItem+Internal.h]: Oh, got it ;)

File: ADAL/src/ADInstanceDiscovery.m (hunk 4 above)
Comment (jasoncoolmax) [inline ADAL/src/ADInstanceDiscovery.m]: Do we need to delete ADInstanceDiscoveryTest.m as well? :)

File: ADAL/tests/ADInstanceDiscoveryTests.m [hunk 5]
```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
```
Comment (jasoncoolmax) [inline ADAL/tests/ADInstanceDiscoveryTests.m]: Do we need to delete ADInstanceDiscoveryTests.m as well? :)

Comment (RPangrle) [inline ADAL/tests/ADInstanceDiscoveryTests.m]: 👍

**Pull request merged**